cdap-security-spi/src/main/java/co/cask/cdap/security/spi/authorization/Authorizer.java

```diff
@@ -64,4 +69,62 @@
    * @param entity the {@link EntityId} on which all {@link Action actions} are to be revoked
    */
   void revoke(EntityId entity);
+
+  /**
+   * Returns all the {@link Privilege} for the specified {@link Principal}.
+   *
+   * @param principal the {@link Principal} for which to return privileges
+   * @return a {@link Set} of {@link Privilege} for the specified principal
+   */
+  Set<Privilege> listPrivileges(Principal principal);
+
+  /********************************* Role Management: APIs for Role Based Access Control ******************************/
+  /**
+   * Create a role.
+   *
+   * @param role the {@link Role} to create
+   * @throws RoleAlreadyExistsException if the the role to be created already exists
+   */
+  void createRole(Role role) throws RoleAlreadyExistsException;
```

Comment:
since these are public APIs that can connect to external systems, we should throw a generic ``Exception`` here. For example, if you were to implement a dataset-backed Authorizer, then there could be ``TransactionFailureException`` in any of these methods - which will have to be wrapped into ``RuntimeException`` if the method contract does not allow any other exceptions. I'm changing the existing APIs to throw ``Exception``, can you make the new Role based ones throw ``Exception`` too?